specifications/api/swagger.yaml

```diff
@@ -2255,6 +2255,39 @@ components:
           <<: *addresses
           description: The recipient addresses.
 
+    ApiWalletUtxoSnapshotEntry: &ApiWalletUtxoSnapshotEntry
+      type: object
+      required:
+        - ada
+        - ada_minimum
+        - assets
+      properties:
+        ada:
+          <<: *amount
+          description: |
+            The ada quantity associated with this UTxO entry.
+        ada_minimum:
+          <<: *amount
+          description: |
+            The minimum ada quantity permitted by the ledger for this UTxO
+            entry.
+        assets:
+          <<: *walletAssets
+          description: |
+            The set of non-ada assets associated with this UTxO entry.
+
+    ApiWalletUtxoSnapshot: &ApiWalletUtxoSnapshot
+      type: object
+      required:
+        - entries
+      properties:
+        entries:
```

Comment:
The extra level of wrapping in an object isn't necessary, is it?

Reply:
> The extra level of wrapping in an object isn't necessary, is it?

I think this is arguable either way.

I agree with you that using a list would be enough to capture the current use case.

But on the other hand, using an object gives us the flexibility to add extra fields relating to a snapshot at a later time, without having to break compatibility with clients of the API (including ourselves).

From an API aesthetics point of view, I feel it's clearer to be able to point to a single type and say "this single type represents a complete UTxO snapshot", similarly to how we use `UTxO` to represent a complete UTxO set.